Recruit-Training/Advanced-Recruit-Training/Embedded-Teaser/Training/Test/TrainingTest.c

```diff
@@ -5,24 +5,37 @@
 void runTrainingTests()
 {
     // Run all the tests here using RUN_TEST()
+    RUN_TEST(test_EverythingValid);
+    RUN_TEST(test_EverythingInvalid);
+    RUN_TEST(test_OnlyLightsInvalid);
+    RUN_TEST(test_OnlyMotorsInvalid);
 }
 
 void test_EverythingValid()
 {
-
+    uint8_t InputArray[3] = {0b00000000, 0b00000000, 0b00000001};
+    trainingTask(InputArray);
+    TEST_ASSERT_EQUAL_MESSAGE(0b00000111, validData, "All valid test failed");
 }
 
 void test_EverythingInvalid()
 {
+    uint8_t InputArray[3] = {0b01000011, 0b010000011, 0b00000011};
+    trainingTask(InputArray);
+    TEST_ASSERT_EQUAL_MESSAGE(0b00000000, validData, "All invalid test failed");
```

Comment:
A better message would be like "data is not blah" that way you know exactly what is wrong when the test fails.

Reply:
So in this case I should put "Data is not 0b00000000"?

Reply:
You could say something like "validData is not 0x0" or "Should not have any valid data". You want to tell the reader what value it _should_ have if the test fails.